components/massmint/EditPanel.vue

```diff
@@ -0,0 +1,198 @@
+              has-counter
+              maxlength="500" />
+            <!-- <div>{{ description.length }}/500</div> -->
+          </o-field>
+          <o-field :label="$t('massmint.price')" class="w-full">
+            <div class="is-flex">
+              <o-input
+                v-model="price"
+                class="is-flex is-flex-grow-2 field-height" />
+              <div
+                class="border-top border-right border-bottom is-flex is-flex-grow-1 field-height is-justify-content-center is-align-items-center">
+                KSM
+              </div>
+            </div>
+          </o-field>
+        </div>
+      </div>
+      <NeoButton
+        class="w-full"
+        :label="$t('massmint.save')"
+        variant="k-accent"
+        :disabled="!name"
+        @click.native="save" />
+    </div>
+  </NeoSidebar>
+</template>
+
+<script setup lang="ts">
+import { NeoButton, NeoIcon, NeoSidebar } from '@kodadot1/brick'
+import { NFT } from './types'
+import { OField, OInput } from '@oruga-ui/oruga'
+const props = defineProps<{
+  nft?: NFT
+  open: boolean
+}>()
+
+const internalNfT = ref<Partial<NFT>>({})
+const name = computed({
+  get: () => internalNfT.value.name || props.nft?.name || '',
+  set: (value) => {
+    internalNfT.value = {
+      ...internalNfT.value,
+      name: value,
+    }
+  },
+})
+const description = computed({
+  get: () => internalNfT.value.description || props.nft?.description || '',
+  set: (value) => {
+    internalNfT.value = {
+      ...internalNfT.value,
+      description: value,
+    }
+  },
+})
+const price = computed({
+  get: () => internalNfT.value.price || props.nft?.price || '',
+  set: (value) => {
+    internalNfT.value = {
+      ...internalNfT.value,
+      price: Number(value),
+    }
+  },
+})
+
+const emit = defineEmits(['close', 'save'])
+
+const closePanel = () => {
+  internalNfT.value = {}
+  emit('close')
+}
+
+const save = () => {
+  const nft = {
+    ...props.nft,
+    ...internalNfT.value,
+  }
+  emit('save', nft)
+  closePanel()
+}
+</script>
+
+<style lang="scss" scoped>
+@import '@/styles/abstracts/variables';
+
+.navbar-margin {
+  margin-top: 83px;
+}
+
+.field-height {
+  height: 3rem;
+
+  :deep input {
+    height: 100%;
+  }
+}
+
+.cover {
+  object-fit: cover;
+}
+
+.sidebar-width:deep .o-side__content {
+  width: 30%;
+}
+
+.form:deep {
+  .o-field__label {
+    margin-bottom: 0.5rem;
+  }
+
+  .o-input__textarea {
+    height: 10rem;
+    padding: 0.5rem;
+
+    @include ktheme() {
+      border: 1px solid theme('text-color');
+      background-color: theme('background-color');
+      color: theme('text-color');
+    }
+  }
+
+  .o-input__counter {
+    position: relative;
+    top: -1.2rem;
+    right: 0.5rem;
+  }
+
+  input {
+    @include ktheme() {
+      border: 1px solid theme('text-color');
+      background-color: theme('background-color');
+      color: theme('text-color');
+    }
+    padding: 0.5rem;
+  }
+}
+
+.o-field--focused:deep {
+  input,
```

Comment:
so many custom css lines

Reply:
yeah well... what would you suggest?
needed some styling, and I'm not sure it justifies creating a NeoInput component

Reply:
> yeah well... what would you suggest? needed some styling, and I'm not sure it justifies creating a NeoInput component

Create `NeoInput` then add [custom css](https://oruga.io/components/Input.html#style)
I'm a bit tired of seeing oruga be used where it should not, I'm pretty sure we'll need this component afterward

Reply:
in libs/ui:
- change props `modelValue` to `value`
- remove vueuse import?
```ts const emit = defineEmits(['input'])
const inputValue = useVModel(props, 'value')
const updateValue = (event) => {
  emit('input', event)
}
```
&
```vue
<template>
  <o-input
    v-model="inputValue"
    class="neo-input"
    v-bind="$attrs"
    :style="heightStyle"
    @input="updateValue" />
</template>
```

This work for me

Reply:
Done

:clinking_glasses:  Thank you